src/server/utils.js

```diff
@@ -0,0 +1,72 @@
+import { ok, error } from '@ucanto/core'
+import { DIDResolutionError } from '@ucanto/validator'
+import { Access as AccessCapabilities, Space as SpaceCapabilities } from '@web3-storage/capabilities'
+import { DelegationFailure } from '../middleware/withDelegationsStorage.types.js'
+import { access } from '@ucanto/validator'
+import { Verifier } from '@ucanto/principal'
+
+
+/**
+ * Checks if the space/content/serve/* delegation is for the gateway and it is not expired.
+ *
+ * @param {import('@ucanto/interface').Signer} gatewayIdentity - The signer of the gateway identity
+ * @param {import('@ucanto/interface').InferInvokedCapability<typeof AccessCapabilities.delegate>} capability - The capability to validate
+ * @param {import('@ucanto/interface').Proof[]} proofs - The proofs to validate
+ */
+export const extractContentServeDelegation = (gatewayIdentity, capability, proofs) => {
+  const nbDelegations = new Set(Object.values(capability.nb.delegations))
+  if (nbDelegations.size > 1) {
+    return { error: new DelegationFailure(`nb.delegations has more than one delegation`) }
+  }
+  const proofDelegations = proofs.flatMap((proof) => 'capabilities' in proof ? [proof] : [])
+  if (nbDelegations.size > proofDelegations.length) {
+    return { error: new DelegationFailure(`nb.delegations has more delegations than proofs`) }
```

Comment:
Redundant as the check below will fail to find some of the delegations in proofs.